cluster/src/dunit/scala/io/snappydata/cluster/QueryRoutingDUnitSecureTest.scala

```diff
@@ -0,0 +1,402 @@
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata.cluster
+
+import java.sql.{BatchUpdateException, Connection, DriverManager, ResultSet, SQLException}
+
+import io.snappydata.test.dunit.AvailablePortHelper
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.collection.Utils
+
+object QueryRoutingDUnitSecureTest{
+  val adminUser: String = "gemfire10"
+}
+
+class QueryRoutingDUnitSecureTest(val s: String)
+    extends ClusterManagerLDAPTestBase(s,
+      QueryRoutingDUnitSecureTest.adminUser) with Logging {
+
+  override def setUp(): Unit = {
+    super.setUp()
+  }
+
+  override def tearDown2(): Unit = {
+    super.tearDown2()
+  }
+
+  def netConnection(netPort: Int, user: String, pass: String): Connection = {
+    val driver = "io.snappydata.jdbc.ClientDriver"
+    Utils.classForName(driver).newInstance
+    val url: String = "jdbc:snappydata://localhost:" + netPort + "/"
+    DriverManager.getConnection(url, user, pass)
+  }
+
+  def insertRows1(numRows: Int, serverHostPort: Int, tableName: String,
+      user: String, pass: String): Unit = {
+    val conn = netConnection(serverHostPort, user, pass)
+    // scalastyle:off println
+    println(s"insertRows1: Connected to $serverHostPort")
+    // scalastyle:on println
+
+    val stmt1 = conn.createStatement()
+    try {
+      var i = 1
+      (1 to numRows).foreach(_ => {
+        stmt1.addBatch(s"insert into $tableName values($i, $i, '$i')")
+        i += 1
+        if (i % 1000 == 0) {
+          stmt1.executeBatch()
+          i = 0
+        }
+      })
+      stmt1.executeBatch()
+
+      // scalastyle:off println
+      println(s"insertRows1: committed $numRows rows")
+      // scalastyle:on println
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+
+  def insertRows2(numRows: Int, serverHostPort: Int, tableName: String,
+      user: String, pass: String): Unit = {
+    val conn = netConnection(serverHostPort, user, pass)
+    // scalastyle:off println
+    println(s"insertRows2: Connected to $serverHostPort")
+    // scalastyle:on println
+
+    val stmt1 = conn.createStatement()
+    try {
+      (1 to numRows).foreach(i => {
+        stmt1.executeUpdate(s"insert into $tableName values($i, $i, '$i')")
+      })
+
+      // scalastyle:off println
+      println(s"insertRows2: committed $numRows rows")
+      // scalastyle:on println
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+
+  def insertRows3(numRows: Int, serverHostPort: Int, tableName: String,
```

Comment:
Looks like the entire code of insert and verify is repeated. Can't we have one in object of one test class and both the test can then share the same code. All the variables anyways you are passing in arguments.